tools/mk_vendor

```diff
@@ -0,0 +1,854 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+NAME="mk_vendor"
+VERSION="0.1"
+
+USAGE="Usage:
+    ${NAME} make <configuration> [vendor-options]
+    ${NAME} locate <configuration> [vendor-options]
+    ${NAME} -h | --help
+    ${NAME} --version
+
+build and interact with vendor libraries
```

Comment:
This needs to be extended to explain in more detail what this tool is used for.

I guess this is expected to be executed from build.rs files, right?

Reply:
It is currently used in the build.rs files of OpenSSL.

However, the longer term idea is to decouple the vendor library build from cargo: the user will first build the vendors he wants (with or without the help of `mk_vendor`) and then call cargo to build tlspuffin.

So, first build whatever you want:
```sh
$ mk_vendor make openssl:openssl111k --name=put1 --options=sancov,asan
$ mk_vendor make openssl:openssl101f --name=put2 --options=sancov,asan --patch=fix-CVE-XYZ
```

Then run:
```sh
$ WITH_PUT=put1,put2 cargo run -p tlspuffin --bin tlspuffin -- ...
```

I agree that this will need a lot more documentation, but I'm waiting for the entire new workflow to be ready before committing to extensive documentation. Also, this might still be subject to changes in the near future.

Reply:
Got it! I think it would be great to have an even more manual process, where you only produce a static Rust library from tlspuffin and then call the linker on your own.

That way it would also possible to distribute binaries and add tooling around it that follow the classical approach of instrumenting/fuzzing.

In AFL++ for instance you have `afl-cc` that you can use to compile your program. After compilation it is no longer a program but a fuzzing target that can be executed. Much like `clang -fsanitize=fuzzer`. If we split all of the steps sufficiently, we can have a `tlspuffin-cc` that you can set using `CC=tlspuffin-cc` and then compile e.g. a patched OpenSSL.

Reply:
I opened #308 to discuss that further. In any case, I agree that we might want to provide smaller build steps at some point, but I think the proposed design matches our current needs without excluding that possibility.